### PR TITLE
feat(eslint): add react plugin

### PR DESCRIPTION
**Pull request by benoitdeziel**
## Description

This PR add a specific eslint config for React.

## Checklist

- [x] Updated the version number in the `package.json`
- [x] Test any new configuration on a repo using [`npm link`](https://docs.npmjs.com/cli/link)

### CODE REVIEW

File: eslint-config-landr/react.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        },
+    },
+    rules: {
+        'react/self-closing-comp': [
+            'error',
+            {
+                component: true,
+                html: false,
+            },
+        ],
```
Comment (henrymyers) [inline eslint-config-landr/react.js]: Nice

File: eslint-config-landr/README.md [hunk 2]
```diff
@@ -13,10 +13,20 @@ Install `eslint-config-landr` and all of it's peer dependencies as dev dependenc
...
 ```
+
+### Framework specific
+
+To use a framework specific config, just use `'landr/${framework}'` in the `extends` array. Every framework specific config already extends the `landr` config.
+
+```js
+module.exports = {
+  extends: ['landr/react'],
+};
```
Comment (ardeois) [inline eslint-config-landr/README.md]: You should specify which framework are supported for now

File: eslint-config-landr/package.json [hunk 3]
```diff
@@ -30,5 +30,8 @@
     },
     "engines": {
         "node": ">= 4"
+    },
+    "dependencies": {
+        "eslint-plugin-react": "^7.16.0"
     }
```
Comment (ardeois) [inline eslint-config-landr/package.json]: we should not add this in dependencies if we don't use react eslint.
Also it forces a version, I think we should let the user choose.
You can use `optionalDependencies` or `peerDependencies` for this

Comment (benoitdeziel) [reply eslint-config-landr/package.json]: true

**Pull request merged**